### PR TITLE
moved inline css and js to assets pipeline

### PR DESCRIPTION
**Pull request by wandtasie**
Hi

I moved some styles and javascripts to assets-pipeline and renamed the files a bit...
I removed the (useless?) var cartFormValidator in https://github.com/jsqu99/spree_flexi_variants/blob/master/app/views/spree/products/_content_for_head.html.erb
and moved the inline style (https://github.com/jsqu99/spree_flexi_variants/blob/master/app/views/spree/shared/_additional_line_item_fields.html.erb) to assets pipeline which got applied multible times (according to the items in cart)

Edith: Not everything is tested but I think there is no problem... normal validation... works

### CODE REVIEW

**Comment by jsqu99** [comment]
thank you!

i will definitely take a look at this asap.

**Comment by wandtasie** [comment]
Most of the other inline styles and js I found use custom code. I will have a look at them next days. I think using the assets pipeline is better because Its so easy to overwrite them.

PS: Thanks for this great extension...

**Comment by wandtasie** [comment]
Hello again

playing around with spree_flexi_variants I got problems with error 500.
https://github.com/jsqu99/spree_flexi_variants/issues/31
I close this pull request and reopen a new one including the changes

**Comment by wandtasie** [comment]
I like Github... but Im new to it...
my changes are already here...

Good night

Matthias

File: app/assets/stylesheets/store/spree-flexi-variants.css [hunk 1]
```diff
@@ -62,4 +62,7 @@ we've got two versions line items: line_items & line-items
 
 #ad_hoc_options {
   margin-top: 20px;
-}
\ No newline at end of file
+}
+table#cart-detail dl {margin-left: 10px;}
```
Comment (wandtasie) [inline app/assets/stylesheets/store/spree-flexi-variants.css:66]: I changed the selectors cause they are now in global file

File: app/assets/javascripts/store/spree_flexi_variants.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  // set up the 'reset' functionality on file uploads
+  $("#cart-form form input[type=file]").each(function () {
+  // make a clone of the original in case the user wants to undo an upload
+    var orig = $(this);
+    var clone = $(orig).clone();
+    attachFileInputSwap(orig, clone);
+  });
+  // image customizations need multipart
+  $("#cart-form form").attr("enctype","multipart/form-data");
+  $("#cart-form form").validate();
```
Comment (wandtasie) [inline app/assets/javascripts/store/spree_flexi_variants.js:15]: I removed the var cartFormValidator, I think we dont need this. do we?

File: app/views/spree/products/_content_for_head.html.erb [hunk 3]
```diff
@@ -59,11 +23,7 @@
       ];
     <% end # javascript_tag%>
 
-    <%= javascript_include_tag 'store/exclusions' %>
+    <%= javascript_include_tag 'store/spree_flexi_variants_exclusions' %>
```
Comment (wandtasie) [inline app/views/spree/products/_content_for_head.html.erb:26]: i renamed the file cause name was to common I think

File: app/views/spree/products/_pricing.html.erb [hunk 4]
```diff
@@ -3,7 +3,6 @@
 %>
 
 <%= content_for :head do %>
-  <%= javascript_include_tag 'jquery.formatCurrency-1.4.0.min.js', 'i18n/jquery.formatCurrency.all.js' %>
```
Comment (wandtasie) [inline app/views/spree/products/_pricing.html.erb:6]: files moved to assets pipeline

File: lib/generators/spree_flexi_variants/install/install_generator.rb [hunk 5]
```diff
@@ -3,14 +3,12 @@ module Generators
     class InstallGenerator < Rails::Generators::Base
 
       def add_javascripts
-        append_file "app/assets/javascripts/store/all.js", "//= require store/exclusions\n"
-        append_file "app/assets/javascripts/store/all.js", "//= require jquery.formatCurrency-1.4.0.min\n"
```
Comment (wandtasie) [inline lib/generators/spree_flexi_variants/install/install_generator.rb:7]: jquery.formatCurrency is in spree_flexy_variants.js manifest, store/exclusions was included twice, now its just included dynamical. maybe we can find a better solution.

File: lib/spree_flexi_variants/engine.rb [hunk 6]
```diff
@@ -24,7 +24,10 @@ def self.activate
     initializer "spree.flexi_variants.preferences", :after => "spree.environment" do |app|
       SpreeFlexiVariants::Config = Spree::FlexiVariantsConfiguration.new
     end
-
+    # store/flexi-variants should be removed after renaming the files
+    initializer "spree.flexi_variants.assets.precompile" do |app|
+        app.config.assets.precompile += ['store/spree_flexi_variants_exclusions.js','store/flexi-variants.*','store/spree-flexi-variants.*']
```
Comment (wandtasie) [inline lib/spree_flexi_variants/engine.rb:29]: this solves error 500 on products page

**Comment by wandtasie** [comment]
bump :-)

@deamonsy verified error 500 on products page in production (I think)
https://github.com/jsqu99/spree_flexi_variants/issues/31#issuecomment-5085819

**Pull request merged**